packages/wonder-blocks-dropdown/src/components/dropdown-core.js

```diff
@@ -310,14 +318,39 @@ class DropdownCore extends React.Component<Props, State> {
         this.removeEventListeners();
     }
 
+    hasSearchBox(): boolean {
+        return (
+            !!this.props.onSearchTextChanged &&
+            typeof this.props.searchText === "string"
+        );
+    }
+
+    // If we have a search box visible, then our focus
+    // index is going to be offset by 1, since the orginal
+    // index doesn't account for the search box's
+    // existence.
+    getIndexOffset(): number {
+        if (this.hasSearchBox()) {
+            return 1;
+        }
+
+        return 0;
+    }
+
     // Figure out focus states for the dropdown after it has changed from open
     // to closed or vice versa
     initialFocusItem() {
         const {initialFocusedIndex, open} = this.props;
 
         if (open) {
             // Reset focused index
-            this.focusedIndex = initialFocusedIndex;
+            // If we are given an initial focus index, select it.  Otherwise
+            // default to the first item
+            if (initialFocusedIndex) {
+                this.focusedIndex = initialFocusedIndex + this.getIndexOffset();
+            } else {
+                this.focusedIndex = 0;
+            }
```

Comment:
We do pretty much the same thing in the constructor.  Can you create a helper method that gets the initial focus index?  `getIndexOffset` should probably be inlined into the new helper.